app/forms/vpc-router-route-edit.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useEffect } from 'react'
+import {
+  useNavigate,
+  type LoaderFunctionArgs,
+  type NavigateFunction,
+} from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQueryClient,
+  usePrefetchedApiQuery,
+  type RouterRouteUpdate,
+} from '@oxide/api'
+
+import { DescriptionField } from '~/components/form/fields/DescriptionField'
+import { ListboxField } from '~/components/form/fields/ListboxField'
+import { NameField } from '~/components/form/fields/NameField'
+import { TextField } from '~/components/form/fields/TextField'
+import { SideModalForm } from '~/components/form/SideModalForm'
+import {
+  fields,
+  routeFormMessage,
+  targetValueDescription,
+} from '~/forms/vpc-router-route/shared'
+import { getVpcRouterRouteSelector, useForm, useVpcRouterRouteSelector } from '~/hooks'
+import { addToast } from '~/stores/toast'
+import { Message } from '~/ui/lib/Message'
+import { pb } from '~/util/path-builder'
+
+EditRouterRouteSideModalForm.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project, vpc, router, route } = getVpcRouterRouteSelector(params)
+  await apiQueryClient.prefetchQuery('vpcRouterRouteView', {
+    path: { route },
+    query: { project, vpc, router },
+  })
+  return null
+}
+
+export function EditRouterRouteSideModalForm() {
+  const queryClient = useApiQueryClient()
+  const routeSelector = useVpcRouterRouteSelector()
+  const { project, vpc, router: routerName, route: routeName } = routeSelector
+  const navigate = useNavigate()
+  const { data: route } = usePrefetchedApiQuery('vpcRouterRouteView', {
+    path: { route: routeName },
+    query: { project, vpc, router: routerName },
+  })
+
+  const defaultValues: RouterRouteUpdate = { ...route }
```

Comment:
So this is kind of interesting. To begin with, I don't think we need to copy this, so you could just do

```ts
const defaultValues: RouterRouteUpdate = route
```

However, there's something else kind of interesting here. Thanks to TypeScript's most problematic characteristic, the `defaultValues` object has a lot more stuff on it at runtime than the type system is aware of. So when you use `body` directly in `onSubmit` without pulling out specific values, you actually get all of them. Somewhat surprisingly, the API does not actually care that there's extra stuff there. I tried it on colo and got a 200. And on reflection, I think if this was a problem, we would have seen it before.

![Screenshot 2024-08-13 at 12 06 53 PM](https://github.com/user-attachments/assets/6c6d7a63-3edd-4af3-97a7-3a1ea786ecdc)

The best thing to do is probably what subnet edit does:

https://github.com/oxidecomputer/console/blob/3a6f815ad50bf2c7fafb55dcdd31ad0232adbfeb/app/forms/subnet-edit.tsx#L53

Though honestly the use of `satisfies` there is probably suboptimal since it doesn't give the resulting thing a type of `VpcSubnetUpdate`. This is probably better:

```ts
const defaultValues: VpcSubnetUpdate = R.pick(subnet, ['name', 'description'])
```

Also based on a quick search (`rg body **/*edit*`) it looks like we usually use destructuring in `onSubmit` to stick only the fields we want in the body.